src/test/java/com/squareup/kotlinpoet/KotlinPoetTest.kt

```diff
@@ -324,6 +325,51 @@ class KotlinPoetTest {
         |""".trimMargin())
   }
 
+  @Test fun extensionFunctionLambdaWithParamName() {
+    val source = FileSpec.builder(tacosPackage, "Taco")
+        .addFunction(FunSpec.builder("whatever")
+            .returns(Unit::class)
+            .receiver(LambdaTypeName.get(
+                parameters = Parameter("name", String::class),
+                returnType = Unit::class.asClassName()))
+            .addStatement("return Unit")
+            .build())
+        .build()
+    assertThat(source.toString()).isEqualTo("""
+      |package com.squareup.tacos
+      |
+      |import kotlin.String
+      |import kotlin.Unit
+      |
+      |fun ((name: String) -> Unit).whatever(): Unit = Unit
+      |""".trimMargin())
+  }
+
+  @Test fun extensionFunctionLambdaWithMultipleParams() {
+    val source = FileSpec.builder(tacosPackage, "Taco")
+        .addFunction(FunSpec.builder("whatever")
+            .returns(Unit::class)
+            .receiver(LambdaTypeName.get(
+                parameters = listOf(
+                    Parameter("name", String::class),
+                    Companion.ofType(Int::class),
```

Comment:
off topic: referencing `Companion` like this seems out of place. Idiomatic?

Reply:
Nope, there seems to be a problem with auto-complete and I didn't notice it, gonna fix.